app/lib/features/chat/widgets/rooms_list.dart

```diff
@@ -307,6 +246,64 @@ class _RoomsListWidgetState extends ConsumerState<RoomsListWidget> {
     );
   }
 
+  List<Widget> renderActions() {
+    final hasFilters = ref.watch(hasRoomFilters);
+    if (_isSearchVisible) {
+      return [
+        Padding(
+          padding: const EdgeInsets.only(right: 8.0),
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.end,
+            children: [
+              TextButton(
+                key: RoomsListWidget.closeSearchActionButtonKey,
+                onPressed: () {
+                  setState(() {
+                    _isSearchVisible = false;
+                  });
+                },
+                child: Text(L10n.of(context).close),
+              ),
```

Comment:
here is an actual code change: we used to have different buttons ("clear" and "close") here depending on whether the user had added any search. I always found that slightly confusing, especially since we have a secondary "clear" just in the search-bar itself. Thus I removed that "clear" behavior from here and replaced it with a "close" that has always the same behavior of only closing ...